torchinductor/ir.py

```diff
@@ -3180,22 +3182,25 @@ def realize_hint(self):
         """
         Called on buffers we expect to be forced to realize later.
         """
-        if self.num_reads() > 1:
+        if isinstance(self.data, (Pointwise, Reduction)) and self.num_reads() > 1:
```

Comment:
When does this ever matter?

Reply:
It just avoids a call to `num_reads`, which is expensive.